### PR TITLE
qt: Add option to uninstall a game.

### PR DESCRIPTION
**Pull request by Steveice10**
Adds an option to the Qt game list to uninstall any installed title. Closes https://github.com/citra-emu/citra/issues/5386

Menu item conditions:
* Everything is enabled if the title is installed, has an installed update, or has an installed DLC.
* Game is enabled if the title is installed.
* Update is enabled if the title has an installed update.
* DLC is enabled if the title has an installed DLC.

New UI elements:

Menus:
Screenshot 2023-10-12 at 2 41 24 AM

Confirmations:
Screenshot 2023-10-12 at 2 41 31 AM
Screenshot 2023-10-12 at 2 41 37 AM
Screenshot 2023-10-12 at 2 41 47 AM
Screenshot 2023-10-12 at 2 41 56 AM

Success:
Screenshot 2023-10-12 at 6 41 31 PM

Failure:
Screenshot 2023-10-12 at 6 43 32 PM

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7064)

### CODE REVIEW

File: src/citra_qt/game_list.cpp [hunk 1]
```diff
@@ -641,8 +668,89 @@ void GameList::AddGamePopup(QMenu& context_menu, const QString& path, u64 progra
...
 
+void GameList::UninstallTitles(
+    const std::vector<std::tuple<Service::FS::MediaType, u64, QString>>& titles) {
+    (void)QtConcurrent::run([this, titles] {
+        for (auto& title : titles) {
+            const auto name = std::get<QString>(title);
+            const auto media_type = std::get<Service::FS::MediaType>(title);
+            const auto program_id = std::get<u64>(title);
+
+            emit main_window->UninstallReport(UninstallStatus::Started, name);
```
Comment (FearlessTobi) [inline src/citra_qt/game_list.cpp:741]: maybe the whole function for removing titles should just be in the main window instead?
then we wouldnt need this helper function

File: src/citra_qt/main.cpp [hunk 2]
```diff
@@ -1750,6 +1751,21 @@ void GMainWindow::OnCIAInstallFinished() {
     game_list->PopulateAsync(UISettings::values.game_dirs);
 }
 
+void GMainWindow::OnUninstallReport(UninstallStatus status, QString name) {
+    switch (status) {
+    case UninstallStatus::Started:
+        this->statusBar()->showMessage(tr("Uninstalling %1").arg(name));
+        break;
+    case UninstallStatus::Succeeded:
+        this->statusBar()->showMessage(tr("%1 has been uninstalled successfully.").arg(name));
```
Comment (FearlessTobi) [inline src/citra_qt/main.cpp:1760]: maybe this should be a popup too?
i don't know if our users will see it otherwise :p

**Review by FearlessTobi — APPROVED** [review]
Thanks for implementing this!

**Comment by Steveice10** [comment]
Addressed comments and reworked the flow a bit. The signals are no longer needed, we use QtConcurrent::map, and it has a progress bar for how many titles have been uninstalled (up to three total possible - game, update, and DLC).

**Pull request merged**